Client/Simitone/Simitone.Client/UI/Panels/UICheatTextbox.cs

```diff
@@ -0,0 +1,209 @@
+                && state.NewKeys.Contains(Keys.C)) //prevent change over multiple frames
+            {
+                Visible = !Visible;
+            }
+            baseTextbox.Visible = Visible;
+            if (Visible)
+            {
+                if (state.NewKeys.Contains(Keys.Enter))
+                {
+                    commandEntered(baseTextbox.CurrentText, out bool shouldHide);
+                    Visible = !shouldHide;
+                }
+            }               
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="commandString"></param>
+        private void commandEntered(string commandString, out bool shouldHide)
+        {
+            shouldHide = true;
+            if (string.IsNullOrWhiteSpace(commandString)) return; // a blank textbox should close after hitting enter -- even if a command was never run.
+
+            var cheat = new VMBudgetCheatCmd();
+            var context = new VMCheatContext();
+            var repetitions = getRepetitions(commandString);
+            switch (trimRepetitions(commandString))
+            {
+                //These three are special cheatcodes that don't really match the parameterized cheats e.g. moveobjects on
+                case "klapaucius":
+                case "rosebud":
+                    context.Amount = (int)VMCheatContext.BudgetCheatPresetAmount.KLAPAUCIUS;
+                    context.CheatBehavior = VMCheatContext.VMCheatType.Budget;
+                    break;
+                case "motherlode":
+                    context.Amount = (int)VMCheatContext.BudgetCheatPresetAmount.MOTHERLODE;
+                    context.CheatBehavior = VMCheatContext.VMCheatType.Budget;
+                    break;
+                default: context = parseCommandString(commandString); break;                                       
+            }
+            cheat.Context = context;
+            shouldHide = false;
+            if (cheat.Context == null) // the command was not recognized
+            {
+                FSO.HIT.HITVM.Get().PlaySoundEvent(UISounds.Error); // in TS1 this was a dialog but a sound may be less intrusive
+                return;
+            }
+            context.Repetitions = repetitions;
+            var sndEvent = UISounds.Error;
+            if (context.CheatBehavior != VMCheatContext.VMCheatType.InvalidCheat)
+            {
+                ts1VM.SendCommand(cheat);
+                switch (context.CheatBehavior) // sound feedback
+                {
+                    case VMCheatContext.VMCheatType.Budget: sndEvent = UISounds.BuyPlace; break;
+                    default: sndEvent = UISounds.Click; break;
+                }
+                shouldHide = true;
+            }
+            FSO.HIT.HITVM.Get().PlaySoundEvent(sndEvent);
+        } 
+
+        private String trimRepetitions(string input)
+        {
+            return input.Replace(";", "").Replace("!", "");
```

Comment:
Remember that this worked because the cheat command system let you use multiple cheats at once, like `move_objects on;rosebud;house 2`. This is fine for now.